api/hasura/cron/recoverTransactions.ts

```diff
@@ -219,21 +225,113 @@ const handleClaim = async (
         },
         { affected_rows: true },
       ],
+      delete_pending_vault_transactions_by_pk: [
+        { tx_hash },
+        { __typename: true },
+      ],
     },
     { operationName: 'recoverClaim' }
   );
 
   return `updated ${update.update_claims?.affected_rows} claims`;
 };
 
+const handleDistribution = async (
+  contracts: Contracts,
+  tx: TxRecord,
+  receipt: TransactionReceipt
+) => {
+  const { distribution_id, tx_hash } = tx;
+  await assertOrRemove(distribution_id, 'no distribution id', tx_hash);
+
+  const rawLog = receipt.logs
+    .filter(log => log.address === contracts.distributor.address)
+    .pop();
+
+  await assertOrRemove(rawLog, 'no event log found', tx_hash);
+  assert(rawLog);
+  const log = contracts.distributor.interface.parseLog(rawLog);
+  await assertOrRemove(
+    log.name === 'EpochFunded',
+    'event name mismatch',
+    tx_hash
+  );
+
+  const { distributions_by_pk: data } = await adminClient.query({
+    distributions_by_pk: [
+      { id: distribution_id },
+      {
+        vault: { id: true, vault_address: true },
+        epoch: { circle_id: true },
+        tx_hash: true,
+        total_amount: true,
+        distribution_json: [{}, true],
+      },
+    ],
+  });
+  assert(data);
+  await assertOrRemove(!data.tx_hash, 'tx_hash already set', tx_hash);
+
+  const previousAmount = BigNumber.from(
+    JSON.parse(data.distribution_json).previousTotal || 0
+  );
+  const amount = BigNumber.from(data.total_amount).sub(previousAmount);
+
+  // these criteria don't strictly uniquely identify a distribution. if
+  // you wanted to be stricter, you could compare timestamps of the tx and
+  // the db row
+  await assertOrRemove(
+    data.vault.vault_address.toLowerCase() === log.args.vault.toLowerCase() &&
+      encodeCircleId(data.epoch.circle_id) === log.args.circle &&
+      amount.toString() === log.args.amount.toString(),
+    'data mismatch',
+    tx_hash
+  );
+
+  const update = await adminClient.mutate(
+    {
+      update_distributions_by_pk: [
+        {
+          pk_columns: { id: distribution_id },
+          _set: { tx_hash, distribution_epoch_id: log.args.epochId.toString() },
+        },
+        { id: true },
+      ],
+      // FIXME: this throws an error about missing Hasura header variables
+      // createVaultTx: [
+      //   {
+      //     payload: {
+      //       tx_type: 'Distribution',
+      //       tx_hash,
+      //       distribution_id,
+      //       circle_id: data.epoch.circle_id,
+      //       vault_id: data.vault.id,
+      //     },
+      //   },
+      //   { id: true },
+      // ],
```

Comment:
This throws an error because Hasura requires an authenticated profile to call this function, so the `created_by` field can be generated. The adminClient doesnt map to a profile so it cannot call this function.